package/server/resource/addCommand/index.ts

```diff
@@ -0,0 +1,133 @@
+import { addAce } from "../acl";
+
+type OxCommandArguments = Record<(string|number), string | number | boolean>;
+
+interface OxCommandParams {
+    name: string;
+    help?: string;
+    paramType?: 'number' | 'playerId' | 'string';
+    optional?: boolean;
+}
+
+interface OxCommandProperties {
+    name?: string;
+    help?: string;
+    params?: OxCommandParams[];
+    restricted?: boolean | string | string[];
+}
+
+let registeredCommmands: OxCommandProperties[] = [];
+let shouldSendCommands = false;
+
+setTimeout(() => {
+    shouldSendCommands = true;
+    emitNet('chat:addSuggestions', -1, registeredCommmands);
+}, 1000)
+
+on('playerJoining', (source: number) => {
+    emitNet('chat:addSuggestions', source, registeredCommmands);
+})
+
+function parseArguments(source: number, args: OxCommandArguments, raw: string, params?: OxCommandParams[]): OxCommandArguments | undefined {
+    if (!params) {
+        return args;
+    }
+
+    let paramMissing = false
+
+    params.every((param, index) => {
+        let arg = args[index]
+        let value: any
```

Comment:
there's no reason not to type this when we are assigning it later and expect it to be of certain type